everest/suite.py

```diff
@@ -169,20 +168,12 @@ def __init__(
         self._shutdown_flag = False  # used to gracefully shut down this thread
         self._error_callback = error_callback
 
-    WAITING_STATUSES = [
-        JobStatus.NOT_ACTIVE,
-        JobStatus.WAITING,
-        JobStatus.SUBMITTED,
-        JobStatus.PENDING,
-    ]
 
     def _cleanup(self) -> None:
         # cleanup
         if self._delete_run_path and self._context is not None:
             for context_index in range(len(self._context)):
-                job_status = self._context.job_status(context_index)
-
-                if job_status == JobStatus.SUCCESS:
+                if self._context.is_job_completed(context_index):
```

Comment:
Does it really check only for the job SUCCESS?

Reply:
Yes. I think we should refactor everest away from using the internal ert states, and expose equivalent methods instead.

Reply:
it looks like Everest needs to know about waiting states, see `suite.py`

Reply:
Yes, I updated both PRs to avoid this by exposing a `is_job_waiting()` method in BatchingContext instead